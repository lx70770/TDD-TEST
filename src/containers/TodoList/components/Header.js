import React from "react";

class Header extends React.PureComponent {
  state = {
    value: ""
  };

  handleInputChange = e => {
    this.setState({ value: e.target.value });
  };

  handleKeyUp = e => {
    const { value } = this.state;
    if (e.keyCode === 13 && value) {
      this.props.addUndoItem(value);
      this.setState({ value: "" });
    }
  };

  render() {
    const { value } = this.state;
    return (
      <div className="Header">
        <div className="Header-content">
          TodoList
          <input
            className="Header-input"
            placeholder="todo"
            onChange={this.handleInputChange}
            onKeyUp={this.handleKeyUp}
            data-test="input"
            value={value}
          />
        </div>
      </div>
    );
  }
}

export default Header;
