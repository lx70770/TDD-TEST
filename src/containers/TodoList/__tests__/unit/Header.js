import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Header from "../../components/Header";

Enzyme.configure({ adapter: new Adapter() });

it("Header渲染正常", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

it("Header组件里包含一个Input框", () => {
  const wrapper = shallow(<Header />);
  const inputElem = wrapper.find("[data-test='input']");
  expect(inputElem.length).toBe(1);
});

it("Header组件Input框初始化内容为空，用户输入时会变化 使用simulate", () => {
  const wrapper = shallow(<Header />);
  const inputElem = wrapper.find("[data-test='input']");
  const userInput = "JEST";
  inputElem.simulate("change", {
    target: { value: userInput }
  });
  expect(wrapper.state("value")).toEqual(userInput);
});

it("Header组件Input框初始化内容为空，用户输入回车 函数不应该被调用", () => {
  const fn = jest.fn();
  const wrapper = shallow(<Header addUndoItem={fn} />);
  const inputElem = wrapper.find("[data-test='input']");
  const userInput = "";
  wrapper.setState({ value: userInput });
  inputElem.simulate("keyUp", {
    keyCode: 13
  });
  expect(fn).not.toHaveBeenCalled();
});

it("Header组件Input框初始化内容不为空，用户输入回车 函数应该被调用", () => {
  const fn = jest.fn();
  const wrapper = shallow(<Header addUndoItem={fn} />);
  const inputElem = wrapper.find("[data-test='input']");
  const userInput = "JEST";
  wrapper.setState({ value: userInput });
  inputElem.simulate("keyUp", {
    keyCode: 13
  });
  expect(fn).toHaveBeenCalled();
  expect(fn).toHaveBeenLastCalledWith(userInput);
});

it("Header组件Input框初始化内容不为空，用户输入回车 input应该被清空", () => {
  const fn = jest.fn();
  const wrapper = shallow(<Header addUndoItem={fn} />);
  const inputElem = wrapper.find("[data-test='input']");
  const userInput = "JEST";
  wrapper.setState({ value: userInput });
  inputElem.simulate("keyUp", {
    keyCode: 13
  });
  const newInputElem = wrapper.find("[data-test='input']");
  expect(newInputElem.prop("value")).toBe("");
});
