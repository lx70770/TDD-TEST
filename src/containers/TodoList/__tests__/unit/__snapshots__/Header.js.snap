// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header渲染正常 1`] = `
<div
  className="Header"
>
  <div
    className="Header-content"
  >
    TodoList
    <input
      className="Header-input"
      data-test="input"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="todo"
      value=""
    />
  </div>
</div>
`;
